src/main/java/edu/ucsb/cs156/courses/jobs/UpdateCourseDataRangeOfQuartersJob.java

```diff
@@ -0,0 +1,131 @@
+package edu.ucsb.cs156.courses.jobs;
+
+import java.util.List;
+import java.util.Optional;
+
+
+import edu.ucsb.cs156.courses.collections.ConvertedSectionCollection;
+import edu.ucsb.cs156.courses.documents.ConvertedSection;
+//import edu.ucsb.cs156.courses.entities.UCSBSubject;
+//import edu.ucsb.cs156.courses.models.Quarter;
+import edu.ucsb.cs156.courses.services.UCSBCurriculumService;
+import edu.ucsb.cs156.courses.services.jobs.JobContext;
+import edu.ucsb.cs156.courses.services.jobs.JobContextConsumer;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+
+@AllArgsConstructor
+@Slf4j
+public class UpdateCourseDataRangeOfQuartersJob implements JobContextConsumer {
+
+    @Getter private String start_quarterYYYYQ;
+    @Getter private String end_quarterYYYYQ;
+    @Getter private UCSBCurriculumService ucsbCurriculumService;
+    @Getter private ConvertedSectionCollection convertedSectionCollection;
+    @Getter private List<String> subjects;
+
+    @Override
+    public void accept(JobContext ctx) throws Exception {
+
+
+        int year1 = Integer.parseInt(start_quarterYYYYQ.substring(0,4));
+        int qtr1 = Integer.parseInt(start_quarterYYYYQ.substring(4));
+
+
+        int year2 = Integer.parseInt(end_quarterYYYYQ.substring(0,4));
+        int qtr2 = Integer.parseInt(end_quarterYYYYQ.substring(4));
+
+        int current_year = year1;
+        int current_qtr = qtr1;
+
+        int end_year = year2;
+        int end_qtr = qtr2;
+    
+        //case 1
+        if(year1 > year2){
+            return;
+            // current_year = year2;
+            // current_qtr = qtr2;
+
+            // end_year = year1;
+            // end_qtr = qtr1;
+        }
+
+        //case 2
+        if(year1 == year2) {
+
+            if(qtr1 > qtr2) {
+                return;
+            }
+            // current_year = year1;
+            // end_year = year2;
+
+            // if(qtr1 <= qtr2){
+            //     current_qtr = qtr1;
+            //     end_qtr = qtr2;
+            // } else {
+            //     current_qtr = qtr2;
+            //     end_qtr = qtr1;
+            // }
+        }
```

Comment:
Tell me more about this code.. 

Also why do we still have commented out code in code that's intended to be merged?